File: archived/src/python/ch01/1.3/l0/remainder.py
def div_identity(n: int, d: int) -> tuple[int, int]:
    """
    Return (q, r) such that n = d*q + r and 0 <= r < |d|.
    Mirrors Python's // and % behavior for integers.
    """
    if d == 0:
        raise ZeroDivisionError("division by zero")
    q, r = divmod(n, d)
    return q, r

def week_shift(start: int, shift: int) -> int:
    """
    Wrap days of week with 0..6 (0=Mon). Return the day index after shift days.
    """
    return (start + shift) % 7

def last_digit_of_power(base: int, exp: int) -> int:
    """
    Compute the last decimal digit of base**exp using modular arithmetic.
    Uses Python's pow with modulus for clarity.
    """
    return pow(base, exp, 10)

File: archived/src/go/ch01/1.3/l0/remainder.go
package main

// DivIdentity returns (q, r) such that n = d*q + r, with r having Go's sign convention.
func DivIdentity(n, d int) (int, int) {
	// caller must ensure d != 0 for meaningful math; Go will panic on division by zero.
	return n / d, n % d
}

// WeekShift wraps 0..6 (0=Mon) and returns the day index after shift days.
// Handles negative inputs robustly.
func WeekShift(start, shift int) int {
	m := 7
	t := ((start % m) + m) % m
	s := ((shift % m) + m) % m
	return (t + s) % m
}

// PowMod computes (base^exp) % mod via binary exponentiation.
func PowMod(base, exp, mod int) int {
	if mod == 1 {
		return 0
	}
	res := 1 % mod
	b := ((base % mod) + mod) % mod
	e := exp
	for e > 0 {
		if e&1 == 1 {
			res = (res * b) % mod
		}
		b = (b * b) % mod
		e >>= 1
	}
	return res
}
